src/sqlfluff/core/rules/functional/segments.py

```diff
@@ -128,6 +128,10 @@ def __getitem__(self, item: int) -> BaseSegment:  # pragma: no cover
     def __getitem__(self, item: slice) -> "Segments":  # pragma: no cover
         pass
 
+    def __iter__(self) -> Iterator[BaseSegment]:
+        # Loops understand we are looping BaseSegment
+        return super().__iter__()
+
```

Comment:
This generally helps with typechecking which is complaining a lot when you actually pull or map elements.
this mapping is often a final pass towards and insert delete etc.

Reply:
Nice!